server/protocols/jmap/src/test/java/org/apache/james/jmap/methods/SetMessagesCreationProcessorTest.java

```diff
@@ -205,9 +223,145 @@ private SetMessagesRequest buildFakeCreationRequest() {
                     .from(DraftEmailer.builder().name("alice").email("alice@example.com").build())
                     .to(ImmutableList.of(DraftEmailer.builder().name("bob").email("bob@example.com").build()))
                     .subject("Hey! ")
-                    .mailboxIds(ImmutableList.of("mailboxId"))
+                    .mailboxIds(ImmutableList.of(OUTBOX_ID))
                     .build()
                 ))
                 .build();
     }
+
+    @Test
+    public void processShouldNotSpoolMailWhenNotSavingToOutbox() throws Exception {
+        // Given
+        Mailbox<TestId> fakeOutbox = mock(Mailbox.class);
+        MessageMapper<TestId> mockMapper = mock(MessageMapper.class);
+        MailboxSessionMapperFactory<TestId> stubSessionMapperFactory = mock(MailboxSessionMapperFactory.class);
+        when(stubSessionMapperFactory.createMessageMapper(any(MailboxSession.class)))
+                .thenReturn(mockMapper);
+        MailSpool mockedMailSpool = mock(MailSpool.class);
+        MailFactory<TestId> mockedMailFactory = mock(MailFactory.class);
+
+        SetMessagesCreationProcessor<TestId> sut = new SetMessagesCreationProcessor<TestId>(null, null,
+                stubSessionMapperFactory, new MIMEMessageConverter(), mockedMailSpool, mockedMailFactory) {
+            @Override
+            protected Optional<Mailbox<TestId>> getOutbox(MailboxSession session) throws MailboxException {
+                TestId stubMailboxId = mock(TestId.class);
+                when(stubMailboxId.serialize()).thenReturn(OUTBOX_ID);
+                when(fakeOutbox.getMailboxId()).thenReturn(stubMailboxId);
+                when(fakeOutbox.getName()).thenReturn("outbox");
+                return Optional.of(fakeOutbox);
+            }
+            @Override
+            protected Optional<TestId> getDraftsId(MailboxSession session) {
+                return Optional.of(TestId.of(1));
```

Comment:
I find hard to read these tests where you mix mocking, stubbing, and overriding.